presto-main/src/test/java/com/facebook/presto/operator/scalar/FunctionAssertions.java

```diff
@@ -629,16 +628,15 @@ else if (type == Block[].class) {
 
     private RowExpression toRowExpression(Session session, Expression projectionExpression)
     {
-        Expression translatedProjection = new SymbolToInputRewriter(INPUT_MAPPING).rewrite(projectionExpression);
-        Map<NodeRef<Expression>, Type> expressionTypes = getExpressionTypesFromInput(
+        Map<NodeRef<Expression>, Type> expressionTypes = getExpressionTypes(
                 session,
                 metadata,
                 SQL_PARSER,
-                INPUT_TYPES,
-                ImmutableList.of(translatedProjection),
+                TypeProvider.copyOf(INPUT_TYPES),
```

Comment:
Wait, why can't this just use `SYMBOL_TYPES`? I'm so confused...

Reply:
This is to replace
```java
Expression translatedProjection = new SymbolToInputRewriter(INPUT_MAPPING).rewrite(projectionExpression);
```

The input channel and symbol mapping will be provided rather through rewrite. This make more sense. Given any `PlanNode`, the source symbol-channel mapping can be easily constructed.

Reply:
I mean, what's the difference between `TypeProvider.copyOf(INPUT_TYPES)` and `SYMBOL_TYPES`? They look exactly the same to me.

Reply:
The reason of this duplication I guess is just because the INPUT_TYPES used to from channel and is changed by different people (Martin & Dain). I should have double checked...good catch :)